ayaneo-platform.c

```diff
@@ -422,73 +368,140 @@ static void ayaneo_led_mc_set(u8 group, u8 pos, u8 brightness)
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 1, color[1]);
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 2, color[2]);
         }
+        ayaneo_led_mc_legacy_set(0x03, 0x00, 0x00);
+}
+
+static void ayaneo_led_mc_legacy_intensity_single(u8 group, u8 *color, u8 zone)
+{
+        ayaneo_led_mc_legacy_set(group, zone, color[0]);
+        ayaneo_led_mc_legacy_set(group, zone + 1, color[1]);
+        ayaneo_led_mc_legacy_set(group, zone + 2, color[2]);
 }
 
 /* KUN doesn't use consistant zone mapping for RGB, adjust */
-static void ayaneo_led_mc_intensity_kun(u8 *color)
-{
-        u8 zone_0 = {3};
-        u8 zone_0_color[3] = {color[1], color[0], color[2]};
-        ayaneo_led_mc_intensity(zone_0_color, 0x03, &zone_0);
-        u8 zone_1 = {6};
-        u8 zone_1_color[3] = {color[1], color[2], color[0]};
-        ayaneo_led_mc_intensity(zone_1_color, 0x03, &zone_1);
-        u8 zone_2 = {9};
-        u8 zone_2_color[3] = {color[2], color[0], color[1]};
-        ayaneo_led_mc_intensity(zone_2_color, 0x03, &zone_2);
-        u8 zone_3 = {12};
-        u8 zone_3_color[3] = {color[2], color[1], color[0]};
-        ayaneo_led_mc_intensity(zone_3_color, 0x03, &zone_3);
-        u8 button_color[3] = {color[2], color[0], color[1]};
-        ayaneo_led_mc_intensity(button_color, 0x04, &zone_3);
+static void ayaneo_led_mc_legacy_intensity_kun(u8 group, u8 *color)
+{
+        u8 zone;
+        u8 remap_color[3];
+
+        if (group == AYANEO_LED_GROUP_BUTTON)
+        {
+                zone = 12;
+                remap_color[0] = color[2];
+                remap_color[1] = color[0];
+                remap_color[2] = color[1];
+                ayaneo_led_mc_legacy_intensity_single(AYANEO_LED_GROUP_BUTTON, remap_color, zone);
+                ayaneo_led_mc_legacy_set(0x03, 0x00, 0x00);
+                return;
+        }
+
+        zone = 3;
+        remap_color[0] = color[1];
+        remap_color[1] = color[0];
+        remap_color[2] = color[2];
+        ayaneo_led_mc_legacy_intensity_single(group, remap_color, zone);
+
+        zone = 6;
+        remap_color[0] = color[1];
+        remap_color[1] = color[2];
+        remap_color[2] = color[0];
+        ayaneo_led_mc_legacy_intensity_single(group, remap_color, zone);
+
+        zone = 9;
+        remap_color[0] = color[2];
+        remap_color[1] = color[0];
+        remap_color[2] = color[1];
+        ayaneo_led_mc_legacy_intensity_single(group, remap_color, zone);
+
+        zone = 12;
+        remap_color[0] = color[2];
+        remap_color[1] = color[1];
+        remap_color[2] = color[0];
+        ayaneo_led_mc_legacy_intensity_single(group, remap_color, zone);
+
+        ayaneo_led_mc_legacy_set(0x03, 0x00, 0x00);
+}
+
+static void ayaneo_led_mc_legacy_off(void)
+{
+        ayaneo_led_mc_legacy_set(0x01, 0x02, 0x31);
+        ayaneo_led_mc_legacy_set(0x02, 0x02, 0x31);
+
+        // note: omit for aya flip when implemented
```

Comment:
Why?

Reply:
I believe the controller on the flip has a bug or quirk which causes unexpected behavior when it receives the commit call.
Updated comments to clarify.